# RAG judge evaluation

Run the current end-to-end hybrid RAG evaluation with the configured LLM judge. It assesses answer correctness and unsupported-question refusals across the 105-case golden dataset; it does not compare prompts or models.

This run makes both answer-generation and judge requests for every golden case. Ensure the agent and judge OpenAI settings are available in the environment or `.env` file before running it.

In [1]:
from pathlib import Path
from subprocess import run

project_root = Path.cwd().resolve()
if not (project_root / 'pyproject.toml').is_file():
    project_root = project_root.parents[1]

run(['make', 'evaluate-judge'], cwd=project_root, check=True)

{
  "dataset": "evaluation/datasets",
  "corpus": "fixtures/corpus",
  "mode": "hybrid",
  "k": 5,
  "metrics": {"cases": 105, "source_precision": 0.9810, "source_recall": 0.9810, "section_precision": 0.9008, "section_recall": 0.9619, "citation_validity": 0.9857, "refusal_accuracy": 0.9524, "error_rate": 0.0},
  "judge": {
    "model": "gpt-4.1-mini",
    "metrics": {"cases": 105, "answer_correct_rate": 0.9714, "mean_judge_latency_ms": 12374.27}
  }
}


**Conclusion:** The judge rated 97.1% of the 105 hybrid RAG cases correct, including the unsupported-question cases, while deterministic metrics showed 98.1% source recall and 98.6% citation validity. This is a strong current baseline, but the 95.2% refusal accuracy and 90.1% section precision identify concrete follow-up work: improve unsupported-question refusal and reduce imprecise section citations. Judge calls add about 12.4 seconds per case before asynchronous overlap, so this evaluation is best used as a periodic quality check.